### ISSUE TITLE
Add error lookup for cli

### ISSUE THREAD

**Issue by andres-lowrie** [sqlfluff/sqlfluff#1136]
Would be neat if one could lookup the error code via the cli, for example 

```shell
sqlfluff view-error L001

[ long output here]
```

Could be another interface for help rather than just the website

**Comment by andres-lowrie** [comment]
I'm gonna try and work on something that parses the `docstring` out of the rules, see what that looks like

**Comment by andres-lowrie** [comment]
so actually looking at the current `rules` command already kinda does this but the `-v` flag doesn't seem to output anything more. That and a positional argument would also be neat to limit the output. Gonna give that a go

**Comment by andres-lowrie** [comment]
So I hacked together a little proof of concept to achieve this. Not sure what the standard operating procedure is for pull requests so in order to not clutter up the notifications on this project, I have the code up in my fork

https://github.com/andres-lowrie/sqlfluff/pull/1/files

I wanted to make sure the approach is correct before writing the tests and all that jazz. Don't know what the thoughts are in regards to the code organization:

In particular: 

https://github.com/andres-lowrie/sqlfluff/pull/1/files#diff-cd2ff6a9be1bdb9da3c94896c319a3cb465db0bc87541fd443467fd0eb46b0d9R151-R154

- should attributes be added to the BaseRule class even though they're only used in cli formatting?
- or is doing this type of "last minute shaping" cool with the project and coding style?

or

https://github.com/andres-lowrie/sqlfluff/pull/1/files#diff-cd2ff6a9be1bdb9da3c94896c319a3cb465db0bc87541fd443467fd0eb46b0d9R186-R187

- this is super crude. I don't know if having a `docstring -> man string` function that transforms the docstring into something more "cli friendly and consistent" even aligns with the roadmap or something this project would be into...

Glad to make any changes to this to get this added.

Thanks in advance, super stoked about this project btw! much needed

**Comment by adam-tokarski** [comment]
@andres-lowrie how it's going? Are you still on that?

**Comment by andres-lowrie** [comment]
I have the fork up and it's working but not sure if I should just open a PR against this project or not. I think I'll just  open a PR in order to make it easier to see diffs

**Comment by pdebelak** [comment]
One note for those that would prefer to use either `man` or `info`, you can use sphinx to generate those docs from this repo. It's likely we could make it easier to install these pre-built, but it is currently possible:

You can build the docs like this:

```
$ git clone https://github.com/sqlfluff/sqlfluff.git # you can select a tag here as well for a specific version
$ cd sqlfluff
$ python3 -m venv .venv
$ . .venv/bin/activate
(.venv) $ pip install -e .
(.venv) $ pip install -r docs/requirements.txt
(.venv) $ make -C docs man info
# optionally install the info files
(.venv) $ make -C docs/build/texinfo install-info
```

You can also install the man page built at `docs/build/man/sqlfluff.1` into your appropriate manpath.
